client/starwhale/api/_impl/service/types/qa.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import inspect
+from typing import Any, Set, List, Callable, Optional
+
+from pydantic import BaseModel
+from pydantic.dataclasses import dataclass
+
+from .types import ServiceType, ComponentSpec
+
+
+@dataclass
+class Message:
+    content: str
+    role: str
+
+
+class Query(BaseModel):
+    user_input: str
+    history: List[Message]
+    confidence: Optional[float]
+    top_k: Optional[float]
+    top_p: Optional[float]
+    temperature: Optional[float]
+    max_length: Optional[int]
+
+
+class QuestionAnswering(ServiceType):
```

Comment:
`QuestionAnswering` -> `LLMQuestionAnswering`
`question_answering` -> `llm_question_answering` ?

The parameters such as top_k, top_p, temperature, etc., are specific and exclusive to language model (LLM) configurations.

Reply:
Changed to `LLMChat`